### ISSUE TITLE
Review Process

### ISSUE THREAD

**Issue by IceDragon200** [GrowthcraftCE/Growthcraft-1.7#34]
After the advent of #31, #32 and #33, I think I need to raise this issue so this problem never happens again. Sorry if I'm picking on anyone, but its for the better.

Basically, what I've been introduced to is the concept of maintainer check offs, where you create PR, the maintainers review it, and then a number of the maintainers give a `LGTM`, "Looks Good To Me", that's when its merged, so PRs are allowed to sit around until a portion of them give the go ahead to merge, therefore the master branch is considered sacred! Thou shall not rebase master, and thou shall not force push over master! This is what feature branches are for, before the branch is merged the maintainers can ask for the commits to be `squashed` to reduce commit footprints, especially if the commits are riddled with small patches/fixups for a feature being added.

So the usual routine is:

```
Create feature branch
Do some work and push commits
Get feature somewhat working
Open PR
Maintainers review and point out things that can be changed/removed/added
Go back do some more work, push commits
Maintainers review again, possibly giving the LGTM, and ask for a squash
Go back squash commits and push
Now everyone waits for tests to pass
ONWARDS TO MERGE CITY
...
PROFIT!?
```

And then you repeat this for every feature, this ensures a level of master branch stability.

I also apologize for nudging the version number (the `2.2.0` release :package: : and `2.2.1` patch :hammer:), that should be done by the maintainers as they deem fit.

Anyway:
tl;dr
When a Pull Request is created, slow down on the merge finger, review it, call in a few more maintainers or community members to review the changes, remember this affects all of us, we should all have input it what goes in and what needs fixing. In short don't be afraid to voice your opinion on a feature.

Here ends my rant :octocat: 

@alatyami 
@Sunconure11

**Comment by Alatyami** [comment]
Completely agree, with one exception, a Pull Request means that development for a given item is done and one is ready for peer review. The only changes that should be made after the pull request is started is to resolve feedback from the PR comments. 

Also, completely agree that pull requests for master should only come from the development branch. So that master (I.e. is never ahead of the development ( I.e. snapshot) and is our golden release copy.

Your PR built successfully and I thought that you were ready from the PR to be merged. It looked good to me, my bad on that one.

We can come up will some development life cycle standards that we, as a dev group, come up with in the wiki.

Side note, I am working on the release process with Travis-CI and pushing the jars back to GitHub and having it build feature branches prior to use completely pull requests.

**Comment by IceDragon200** [comment]
Yeah I was peeking at the releases and noticed a new commit on master earlier as well, how the jar pushing going?

**Comment by Alatyami** [comment]
Can you review https://github.com/alatyami/Growthcraft-1.7/wiki/Contributing-to-Growthcraft I added a review process section. 

I think we are going to drop the modular jar and just stick with the single jar distro. I'm not seeing much positive feedback on the forums for splitting them out again.

**Comment by Ar97x** [comment]
I like the idea, mostly because I can add bamboo to new structures, or
renew them and add a little more detail. Will we keep it all at one?

**Comment by IceDragon200** [comment]
Single jars make it easier to update, but then you'll drag in modules you probably didn't want, we could keep both, making the modular jars lower priority

**Comment by Alatyami** [comment]
Gwafu used some kind of voodoo magic with exclusions that even he said had a hard time to make work. If we are going to split them out then we will need to do it right and split the project up.

**Comment by IceDragon200** [comment]
I had a ruby script which did that voodoo ;O

_points at tools/package.rb_

**Comment by Alatyami** [comment]
Doh ... need a new set of glasses. Thanks for the reference! I'm closing out this issue because this issue wanted a review process. We one and #87 is to review that establish that process.
